### ISSUE TITLE
move shared "read" and types code from relayer/conductor to utils crate

### ISSUE THREAD

**Issue by noot** [astriaorg/astria#27]
currently, there is a decent amount of code shared between the relayer and conductor, mainly by means of the conductor importing the relayer code. however, this shared code should live in some utils crate used by both. this will allow them to not depend on each other, and also facilitate testing within the conductor.

**Comment by noot** [comment]
will do this after #99

**Comment by joroshiba** [comment]
@SuperFluffy is this related to your current work?

**Comment by SuperFluffy** [comment]
@joroshiba yeah, that's all the same issue essentially

**Comment by noot** [comment]
closing as duplicate of #86
